### PR TITLE
Fix adding/removing bug

### PR DESCRIPTION
**Pull request by AbhiPrasad**
So we mutated react state previously when adding/removing activities. This sometimes makes it a mess when adding/removing and jumping back and forth between screens quickly. I just made it so that we create new arrays when we setState.

Also I changed up the reducer to return state when we add so that things update correctly when we jump between my schedule and all activities.

### CODE REVIEW

File: app/components/activity/ActivityList.js [hunk 1]
```diff
@@ -169,7 +165,6 @@ class ActivityList extends React.Component {
 		this.props.navigation.navigate('ActivityDetails', {
 			index,
 			currentActivity: activity,
-			activitiesList: this.state.currentActivities,
```
Comment (AbhiPrasad) [inline app/components/activity/ActivityList.js]: Removed this because it's not being used at all

File: app/components/activity/ActivityList.js [hunk 2]
```diff
@@ -63,19 +63,15 @@ class ActivityList extends React.Component {
 
 	onAddButtonPress(itemId) {
 		const { userId, myActivities } = this.state;
-		if (myActivities.indexOf(itemId) >= 0) return;
-
-		myActivities.push(itemId);
-		this.state.onAddActivity(userId, myActivities);
+		this.setState({ myActivities: [...myActivities, itemId] })
```
Comment (theRoughCode) [inline app/components/activity/ActivityList.js]: Would this prevent duplicated ids?

File: app/components/activity/MySchedule.js [hunk 3]
```diff
@@ -65,19 +71,15 @@ class MySchedule extends React.Component {
 
 	onAddButtonPress(itemId) {
 		const { userId, myActivities } = this.state;
-		if (myActivities.indexOf(itemId) >= 0) return;
-
-		myActivities.push(itemId);
-		this.state.onAddActivity(userId, myActivities);
+		this.setState({ myActivities: [...myActivities, itemId] });
```
Comment (theRoughCode) [inline app/components/activity/MySchedule.js]: Duplicate ids?

File: app/components/activity/MySchedule.js [hunk 4]
```diff
@@ -120,13 +122,12 @@ class MySchedule extends React.Component {
 		);
   }
 
-  renderActivityDetails(activity, index) {
+  renderActivityDetails(activity) {
 		// Due to SectionList, the passed-in index is for each section (incorrect)
-		index = this.state.currentActivities.indexOf(activity);
+		const index = this.state.currentActivities.indexOf(activity);
```
Comment (theRoughCode) [inline app/components/activity/MySchedule.js]: 👍

File: app/components/activity/ActivityList.js (hunk 2 above)
Comment (AbhiPrasad) [reply app/components/activity/ActivityList.js]: No it would not iirc

File: app/reducers.js [hunk 5]
```diff
@@ -97,8 +97,9 @@ const myActivities = (state = [], action) => {
 		case USER_ACTIVITY_LOADED:
 			return action.payload.activities;
 		case ADD_ACTIVITY:
+			return state;
```
Comment (theRoughCode) [inline app/reducers.js]: Why aren't we return the new array of activities?

Comment (AbhiPrasad) [reply app/reducers.js]: yeah, its probably best to just form the new array in the reducer. Right now the function is set up when you dispatch `add activity` it gives it the new array, which is probably not the best `redux` way to do it. I'll fix it up!

**Comment by AbhiPrasad** [comment]
Closing because of conflicts, opening up new PR

**Pull request closed without merging**